server/src/main/java/org/opensearch/search/profile/ProfileResult.java

```diff
@@ -105,6 +137,23 @@ public ProfileResult(StreamInput in) throws IOException {
         breakdown = in.readMap(StreamInput::readString, StreamInput::readLong);
         debug = in.readMap(StreamInput::readString, StreamInput::readGenericValue);
         children = in.readList(ProfileResult::new);
+        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+            this.concurrent = in.readBoolean();
+            if (concurrent) {
+                this.maxSliceNodeTime = in.readLong();
+                this.minSliceNodeTime = in.readLong();
+                this.avgSliceNodeTime = in.readLong();
+            } else {
+                this.maxSliceNodeTime = this.nodeTime;
+                this.minSliceNodeTime = this.nodeTime;
+                this.avgSliceNodeTime = this.nodeTime;
+            }
+        } else {
+            this.concurrent = false;
+            this.maxSliceNodeTime = this.nodeTime;
```

Comment:
Curious if this would be better (no measurements):
```suggestion
            this.maxSliceNodeTime = -1;
            this.minSliceNodeTime = -1;
            this.avgSliceNodeTime = -1;
```

Reply:
Thanks, @reta. It has been addressed now.